indexing-hadoop/src/main/java/io/druid/indexer/DetermineHashedPartitionsJob.java

```diff
@@ -324,12 +324,9 @@ protected void reduce(
             HyperLogLogCollector.makeCollector(ByteBuffer.wrap(value.getBytes(), 0, value.getLength()))
         );
       }
-      Optional<Interval> intervalOptional = config.getGranularitySpec().bucketInterval(new DateTime(key.get()));
 
-      if (!intervalOptional.isPresent()) {
-        throw new ISE("WTF?! No bucket found for timestamp: %s", key.get());
-      }
-      Interval interval = intervalOptional.get();
+      Interval interval = config.getGranularitySpec().getSegmentGranularity().bucket(new DateTime(key.get()));
```

Comment:
You can replace `config.getGranularitySpec().bucketInterval()` with `config.getGranularitySpec().getSegmentGranularity().bucket()` in this class and it passes the unit tests. The main difference is `bucketInterval` can return empty when there are no intervals but `bucket` will work for all cases. I don't know the historical context for why `bucketInterval` was added to the GranularitySpec interface.